src/client/page.ts

```diff
@@ -451,11 +451,19 @@ export class Page extends ChannelOwner<channels.PageChannel, channels.PageInitia
   }
 
   async close(options: { runBeforeUnload?: boolean } = {runBeforeUnload: undefined}) {
-    return this._wrapApiCall('page.close', async () => {
-      await this._channel.close(options);
-      if (this._ownedContext)
-        await this._ownedContext.close();
-    });
+    try {
+      await this._wrapApiCall('page.close', async () => {
+        await this._channel.close(options);
+        if (this._ownedContext)
+          await this._ownedContext.close();
+      });
+    } catch (e) {
+      if (e.message === 'page.close: Browser has been closed')
+        return;
+      if (e.message === 'page.close: Target browser or context has been closed')
```

Comment:
This is rather brittle, should we just use a special class for such errors/mark them with a symbol?

Reply:
I can do a shared constant for this.